From: Peter Wu <lekensteyn@gmail.com>
Date: Wed, 25 Sep 2013 22:55:55 +0000 (+0200)
Subject: Fix compile warning about unused typedef
X-Git-Url: http://anonscm.debian.org/gitweb/?p=collab-maint%2Fltrace.git;a=commitdiff_plain;h=a77b267abb870c935f9b4187ca31409fa5eb9dcb

Fix compile warning about unused typedef

Switch assertion to style as used by the Linux kernel (BUILD_BUG_ON)

Signed-off-by: Peter Wu <lekensteyn@gmail.com>
---

diff --git a/lens_default.c b/lens_default.c
index e0c0566..84b74ea 100644
--- a/lens_default.c
+++ b/lens_default.c
@@ -345,7 +345,7 @@ format_array(FILE *stream, struct value *value, struct value_dict *arguments,
 {
 	/* We need "long" to be long enough to cover the whole address
 	 * space.  */
-	typedef char assert__long_enough_long[-(sizeof(long) < sizeof(void *))];
+	(void)sizeof(char[1 - 2*(sizeof(long) < sizeof(void *))]);
 	long l;
 	if (expr_eval_word(length, value, arguments, &l) < 0)
 		return -1;
diff --git a/value.c b/value.c
index d18db17..8caf98c 100644
--- a/value.c
+++ b/value.c
@@ -284,7 +284,7 @@ value_init_deref(struct value *ret_val, struct value *valp)
 
 	/* We need "long" to be long enough to hold platform
 	 * pointers.  */
-	typedef char assert__long_enough_long[-(sizeof(l) < sizeof(void *))];
+	(void)sizeof(char[1 - 2*(sizeof(l) < sizeof(void *))]);
 
 	value_common_init(ret_val, valp->inferior, valp,
 			  valp->type->u.ptr_info.info, 0);
